pkg/utils/kubernetes/referencegrant.go

```diff
@@ -0,0 +1,67 @@
+package kubernetes
+
+import (
+	"context"
+
+	"github.com/samber/lo"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+// AllowedByReferenceGrants checks if the reference from the input `from` to the object(s)
+// in namespace `targetNamespace` with group, kind, name given in the input `to` is allowed by any ReferenceGrant.
+func AllowedByReferenceGrants(
```

Comment:
Please consider reusing this new generic function in https://github.com/Kong/gateway-operator/blob/3d5b5061ad25cba0d261e75bc88a6d75857d0a9d/controller/pkg/secrets/ref/ref.go#L75 to limit logic duplication.

Reply:
I think it would make code clearer and more extensible if we refactor `CheckReferenceGrantForSecret` to call `AllowedByReferenceGrants` here. There will be more usage of checking reference permissions by `ReferenceGrant` between other kinds of resources.